### ISSUE TITLE
General purpose dialog widget

### ISSUE THREAD

**Issue by mariotaku** [kisvegabor/lvgl#2560]
### Introduce the problem

We have `lv_msgbox`, which is a good widget to display a modal popup. It would be better to be more customizable to add something else, like a `lv_textarea` for example.

### Examples and cases

Desired result:

![image](https://user-images.githubusercontent.com/830358/133634561-ef48ace1-f3c4-4440-b823-bb4be91dc946.png)

### Suggested solution

This is already implemented. I'll create PR shortly after #2484.

**Comment by rzr** [comment]
Sounds nice, Isn't this an extra widget built on core ones ?


FYI, I have tried my own for:

https://forum.lvgl.io/t/dialog-lvgl-ui-utility-to-use-along-scripts/6808


![https://git.ostc-eu.org/rzr/dialog-lvgl](https://forum.lvgl.io/uploads/default/original/2X/4/4a95749b4eb5a107d2f2f7ea17ea3a92b026f3fa.png)

**Comment by mariotaku** [comment]
This is a modified version of `lv_msgbox`. And they share similar styles.

**Comment by mariotaku** [comment]
Good work! Looks like yours is a standalone tool for other apps. While mine is a widget used inside LVGL based app itself.

**Comment by kisvegabor** [comment]
What about considering `lv_msgbox` as a modular widget with functions like
```c
lv_msgbox_add_textarea();
lv_msgbox_add_img();
lv_msgbox_add_label();
lv_msgbox_add_container(); 
lv_msgbox_add_btnmatrix(); 
etc..
```

All elements would have the same width as the message box (width = 100%), and placed below each other.

**Comment by mariotaku** [comment]
Currently it looks like `lv_win`. It has `lv_dialog_get_content()`. Yours is also a good option but this means user have to set styles of common parts on their own (btnmatrix etc). What do you think?

**Comment by rzr** [comment]
I meant should those new widgets be optional to keep lvgl smaller as possible ? that said I wont be against reimplementing this extra dialog sample app with more advanced widgets, I'll keep an eye on it.

**Comment by mariotaku** [comment]
@rzr Don't worry. Each widget are controlled by macro to turn it on or off. If it's not enabled, it will not be included into target binary.

**Comment by kisvegabor** [comment]
> Currently it looks like lv_win

What is the added value of `lv_dialog` compared to `lv_win` and manually adding arbitrary widgets?

**Comment by mariotaku** [comment]
In case you're curious: https://github.com/mariotaku/lvgl/compare/sdl-renderer..lv_dialog

~~If you're okay with merging #2484, I'll create a new PR about this one (although has nothing to do with GPU code, that branch is based on it). And let's talk about the design there.~~
I'll open a PR for this :)
